lib/src/components/NxFilterInput/NxFilterInput.tsx

```diff
@@ -23,16 +26,39 @@ const NxFilterInput = forwardRef<HTMLDivElement, Props>(
           className = classnames('nx-filter-input', classNameProp, {
             'nx-filter-input--empty': isEmpty
           }),
-
+          btnClassName = classnames('nx-btn--clear', { 'hidden': isEmpty }),
+          fieldRef = useRef<HTMLDivElement>(null),
+          mergedRef = useMergedRef(fieldRef, ref),
           // just in case these props get passed in, avoid passing them to NxTextInput as they would cause
           // malfunction
           cleanedProps = omit(['validatable', 'validationErrors', 'type'], otherProps),
           filterIcon = searchIcon ? faSearch : faFilter,
-          prefixContent = <NxFontAwesomeIcon icon={filterIcon} className="nx-icon--filter-icons" />;
+          prefixContent = <NxFontAwesomeIcon icon={filterIcon} className="nx-icon--filter-icons" />,
+          suffixContent =
+            <NxButton className={btnClassName}
+                      variant="icon-only"
+                      title="clear filter"
+                      onClick={clearFilterInputText}
+                      tabIndex={-1}>
+              <Close/>
+            </NxButton>;
+
+      function clearFilterInputText() {
+        if (props.onChange) {
+          props.onChange('');
+        }
+      }
+
+      const handleKeyDown: KeyboardEventHandler<HTMLInputElement | HTMLTextAreaElement> = (e) => {
+        if (e.key === 'Escape') {
+          clearFilterInputText();
```

Comment:
Should this also do a `preventDefault`? I'm imagining if we have something like a filter input inside of a modal, we don't want Escape here to close the modal. `NxDropdown` has similar logic.

Reply:
> Should this also do a preventDefault? I'm imagining if we have something like a filter input inside of a modal, we don't want Escape here to close the modal. NxDropdown has similar logic.

In fact let's add one of these to the modal form example and (once we've decided how it should behave) test that it behaves correctly there.

Reply:
I test it in the modal form example and it closes the modal when Escape is pressed, also notice it is mentioned on the modal page doc that any components within the modal that have their own ESC handling should call `preventDefault` on any ESC keydowns that they handle in order to prevent the modal from also handling them. 
Updated here 22e4fcdbfb6fdfc0dfbfc3c5cb87df9bd5c49cd2. Thanks for catching this.